diff --git a/plugins/CMakeLists.txt b/plugins/CMakeLists.txt
index e07a10c5b..c37551df4 100644
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -9,6 +9,7 @@ add_subdirectory(producer_api_plugin)
 add_subdirectory(history_plugin)
 add_subdirectory(history_api_plugin)
 add_subdirectory(state_history_plugin)
+add_subdirectory(producer_heartbeat_plugin)
 
 add_subdirectory(wallet_plugin)
 add_subdirectory(wallet_api_plugin)
diff --git a/programs/nodeos/CMakeLists.txt b/programs/nodeos/CMakeLists.txt
index d5fe8273e..0444fcbe5 100644
--- a/programs/nodeos/CMakeLists.txt
+++ b/programs/nodeos/CMakeLists.txt
@@ -61,6 +61,7 @@ target_link_libraries( ${NODE_EXECUTABLE_NAME}
         PRIVATE -Wl,${whole_archive_flag} db_size_api_plugin         -Wl,${no_whole_archive_flag}
         PRIVATE -Wl,${whole_archive_flag} producer_api_plugin        -Wl,${no_whole_archive_flag}
         PRIVATE -Wl,${whole_archive_flag} test_control_plugin        -Wl,${no_whole_archive_flag}
+        PRIVATE -Wl,${whole_archive_flag} producer_heartbeat_plugin  -Wl,${no_whole_archive_flag}
         PRIVATE -Wl,${whole_archive_flag} test_control_api_plugin    -Wl,${no_whole_archive_flag}
         PRIVATE -Wl,${build_id_flag}
         PRIVATE chain_plugin http_plugin producer_plugin http_client_plugin
